File: datacollector/src/main/java/com/example/yunlong/datacollector/utils/TimeUtils.java
package com.example.yunlong.datacollector.utils;

import android.util.Log;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

/**
 * Created by Yunlong on 4/24/2016.
 */
public class TimeUtils {

    public static String getCurrentTimeStr(){
        Calendar c = Calendar.getInstance();
        SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd:HH-mm-ss");
        String currentTimeStr = df.format(c.getTime());
        return currentTimeStr;
    }

    public static Date getDateFromString(String timeStr){
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd:HH-mm-ss");
        Date d = new Date();
        try {
            d = sdf.parse(timeStr);
        } catch (ParseException ex) {
            Log.d("TimeUtils",ex.getMessage());
        }
        return d;
    }
}

File: datacollector/src/main/java/de/leo/smartTrigger/datacollector/utils/TimeUtils.kt
package de.leo.smartTrigger.datacollector.utils

import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneId
import org.threeten.bp.ZonedDateTime
import org.threeten.bp.format.DateTimeFormatter
import java.util.concurrent.TimeUnit

/**
 * Created by Leo on 4/24/2016.
 */
object TimeUtils {

    val currentTimeStr: String
        get() = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))

    fun getTimeStr(date: LocalDateTime): String {
        return date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))
    }

    fun getDateFromString(timeStr: String): ZonedDateTime {
        return LocalDateTime.parse(timeStr, DateTimeFormatter.ofPattern("yyyy-MM-dd:HH-mm-ss"))
            .atZone(ZoneId.systemDefault())
    }

    fun ZonedDateTime.toEpochMillis(): Long = TimeUnit.SECONDS.toMillis(toEpochSecond())
}
